Build failure with CONFIG_NET_DEBUG_APP=y

Sample applications fail to build on defining CONFIG_NET_DEBUG_APP=y in project configuration.
Error log with zephyr/samples/net/lwm2m_client/ is given below.

-------------------------------------------------------------------------------------------------------------
  LINK    zephyr.lnk
subsys/built-in.o: In function `net_app_connect':
/home/b16394/network_protocols/networkprotocols/WP1/zephyr/subsys/net/lib/app/client.c:599: undefined reference to `_net_app_select_net_ctx'
collect2: error: ld returned 1 exit status
make[2]: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/Makefile:876: zephyr_prebuilt.elf] Error 1
make[2]: Leaving directory '/home/b16394/network_protocols/networkprotocols/WP1/zephyr/samples/net/lwm2m_client/outdir/qemu_x86'
make[1]: *** [Makefile:178: sub-make] Error 2
make[1]: Leaving directory '/home/b16394/network_protocols/networkprotocols/WP1/zephyr'
make: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/Makefile.inc:82: all] Error 2

--- a/subsys/net/lib/app/net_app.c
+++ b/subsys/net/lib/app/net_app.c
@@ -680,9 +680,6 @@ struct net_context *select_server_ctx(struct net_app_ctx *ctx,
 #endif /* CONFIG_NET_APP_SERVER */
 
 #if NET_LOG_ENABLED > 0
-#define _net_app_select_net_ctx(ctx, dst)				\
-	_net_app_select_net_ctx_debug(ctx, dst, __func__, __LINE__)
-
 struct net_context *_net_app_select_net_ctx_debug(struct net_app_ctx *ctx,
 						  const struct sockaddr *dst,
 						  const char *caller,

--- a/subsys/net/lib/app/net_app_private.h
+++ b/subsys/net/lib/app/net_app_private.h
@@ -77,8 +77,19 @@ int _net_app_config_local_ctx(struct net_app_ctx *ctx,
 			      enum net_sock_type sock_type,
 			      enum net_ip_protocol proto,
 			      struct sockaddr *addr);
+
+#if NET_LOG_ENABLED > 0
+struct net_context *_net_app_select_net_ctx_debug(struct net_app_ctx *ctx,
+						  const struct sockaddr *dst,
+						  const char *caller,
+						  int line);
+#define _net_app_select_net_ctx(ctx, dst)				\
+	_net_app_select_net_ctx_debug(ctx, dst, __func__, __LINE__)
+#else
 struct net_context *_net_app_select_net_ctx(struct net_app_ctx *ctx,
 					    const struct sockaddr *dst);
+#endif
+
 int _net_app_ssl_mux(void *context, unsigned char *buf, size_t size);
 int _net_app_tls_sendto(struct net_pkt *pkt,
 			const struct sockaddr *dst_addr,

